### PR TITLE
Implement bep-127 and bep-131 (#131)

### PR DESCRIPTION
**Pull request by cosinlink**
### [BEP-127: Temporary Maintenance Mode for Validators](https://github.com/bnb-chain/BEPs/pull/127)
- This BEP introduces the Temporary Maintenance mode for validators on the BNB Smart Chain.

### [BEP-131: Introduce candidate validators onto BNB Smart Chain](https://github.com/bnb-chain/BEPs/pull/131)
- This BEP introduces candidate validators onto BNB Smart Chain to improve the liveness and robustness of the network.

### Fix bugs
- fix bug that validatorSet order was changed incorrectly while `_felony` in `_forceMaintainingValidatorsExit`, using numOfCabinets not init
- fix bug that slash count calculation problem while updating validator set

### CODE REVIEW

**Review by guagualvcha — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -14,28 +14,56 @@
...
...000000000000000aea1ac326886b992a991d21a6eb155f41b77867cbf659e78f31d89d8205122a84d1be64f0e9a466c2e66a53433928192783e29f8fa21beb2133499b5ef770f60000000e8d4a5100099308aa365c40554bc89982af505d85da95251445d5dd4a9bb37dd2584fd92d3000000e8d4a5100001776920ff0b0f38d78cf95c033c21adf7045785114e392a7544179652e0a612000000e8d4a51000 --initValidatorSetBytes f901a880f901a4f844941284214b9b9c85549ab3d2b972df0deef66ac2c9946ddf42a51534fc98d0c0a3b42c963cace8441ddf946ddf42a51534fc98d0c0a3b42c963cace8441ddf8410000000f84494a2959d3f95eae5dc7d70144ce1b73b403b7eb6e0948081ef03f1d9e0bb4a5bf38f16285c879299f07f948081ef03f1d9e0bb4a5bf38f16285c879299f07f8410000000f8449435552c16704d214347f29fa77f77da6d75d7c75294dc4973e838e3949c77aced16ac2315dc2d7ab11194dc4973e838e3949c77aced16ac2315dc2d7ab1118410000000f84494980a75ecd1309ea12fa2ed87a8744fbfc9b863d594cc6ac05c95a99c1f7b5f88de0e3486c82293b27094cc6ac05c95a99c1f7b5f88de0e3486c82293b2708410000000f84494f474cf03cceff28abc65c9cbae594f725c80e12d94e61a183325a18a173319dd8e19c8d069459e217594e61a183325a18a173319dd8e19c8d069459e21758410000000f84494b71b214cb885500844365e95cd9942c7276e7fd894d22ca3ba2141d23adab65ce4940eb7665ea2b6a794d22ca3ba2141d23adab65ce4940eb7665ea2b6a78410000000",
```
Comment (guagualvcha) [inline package.json:21]: generate-testnet is different?

**Review by guagualvcha — CHANGES_REQUESTED** [review]
Need double check

File: package.json (hunk 1 above)
Comment (cosinlink) [reply package.json:21]: due to the `,` on the end of the line

Comment (cosinlink) [reply package.json:21]: I will modify the line of `generate-QA` to reduce this concern

Comment (guagualvcha) [reply package.json:21]: ok, let us move `generate-QA` upper.

Comment (cosinlink) [reply package.json]: fixed

**Review by guagualvcha — APPROVED** [review]

**Review by yutianwu — APPROVED** [review]

**Review by j75689 — APPROVED** [review]

**Pull request merged**